peakina/readers/excel.py

```diff
@@ -0,0 +1,199 @@
+"""
+Module to add xml support
+"""
+import enum
+from io import StringIO
+from typing import Any, Generator, List, Optional, Tuple, Union
+
+import pandas as pd
+import xlrd
+from openpyxl import load_workbook
+from openpyxl.utils.exceptions import InvalidFileException
+
+from peakina.readers.common import PreviewArgs
+
+
+class EXCEL_TYPE(enum.Enum):
+    NEW = "new"
+    OLD = "old"
+
+
+def _old_xls_rows_iterator(
+    wb: Any, sh_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    if preview:
+        to_iter = range(preview.offset, preview.offset + preview.nrows)
+    else:
+        to_iter = range(wb[sh_name].nrows)
+
+    for rx in to_iter:
+        yield wb[sh_name].row(rx)
+
+
+def _new_xls_rows_iterator(
+    wb: Any, sh_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    yield wb[sh_name].iter_rows(
+        min_row=preview.offset if preview else None,
+        max_row=preview.offset + preview.nrows if preview else None,
+        values_only=True,
+    )
+
+
+def _get_rows_iterator(
+    wb: Any, excel_type: EXCEL_TYPE, sheet_name: str, preview: Optional[PreviewArgs]
+) -> Generator[Any, Any, Any]:
+
+    if excel_type is EXCEL_TYPE.OLD:
+        return _old_xls_rows_iterator(wb, sheet_name, preview)
+
+    return _new_xls_rows_iterator(wb, sheet_name, preview)
+
+
+def _build_row_subset(
+    row: Union[List[Any], Tuple[Any]],
+    sh_name: str,
+    sheetnames: List[str],
+    row_number: int,
+    row_subset: List[str],
+) -> Tuple[int, List[str]]:
+
+    cells = [str(cell.value) if type(cell) not in [str, int, float] else str(cell) for cell in row]
```

Comment:
:question: 

How does hat goes with types like boolean and dates?
Are we able to write values here that will be interpreted as the correct type by `pandas.read_csv`?

My guess is that, for booleans, excel stores these as "TRUE" and "FALSE", so maybe we should add these into the `true_values` and the `false_values` paramaeters of read_csv.

For dates, as we need to encode something as string, we may need to save which columns are supposed to contain dates, and then use the `parse_dates` parameter with the list of these columns.

Reply:
Thanks for your comment,
You're right, i think it should be something like : 
```suggestion
    cells = [cell.value if isinstance(cell, Cell) else str(cell) for cell in row]
```
With Cell, an object type from  https://openpyxl.readthedocs.io/en/stable/api/openpyxl.cell.cell.html on openpyxl or https://xlrd.readthedocs.io/en/latest/api.html?highlight=cell#xlrd.sheet.Sheet.cell for xldr.

What do you think ?

Reply:
It's a bit more complicated than just this.

Lets take an example. Here is a spreadsheet:

| label | value | date | valid |
|-|-|-|-|
| A | 1 | 1/2/2021 | TRUE |
| B | 2 | 1/3/2021 | FALSE |
| C | 3 | 1/4/2021 | TRUE |

parsing with pandas.read_excel should lead to a dataframe where the dtype of column are:
- `string` for label
- `int` for value
- `datetime` for date
- `boolean` for valid

However, now that we have a conversion to csv in the process, we'll call pandas.read_csv on something like:
```
label,value,date,valid
A,1,1/2/2021,True
B,2,1/3/2021,False
C,3,1/4/2021,True
```
How to make sure then than the valid and the date columns have the correct dtype in the end?

I don't know the anwser yet the best would be to include such cases in the test set so we know the current behavior, and see if it changes with this patch.